### PR TITLE
Add ground projection support

### PR DESCRIPTION
**Pull request by sebavan**
https://forum.babylonjs.com/t/projected-ground-skybox/40568

can be tested with https://playground.babylonjs.com/#XG08YC

related doc PR https://github.com/BabylonJS/Documentation/pull/1000

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14470/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14470/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14470/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14470/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14470/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14470/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by Popov72 — APPROVED** [review]
Great!

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Shaders/background.fragment.fx [hunk 1]
```diff
@@ -108,6 +106,84 @@ varying vec3 vNormalW;
...
+    }
+
+    vec3 project(vec3 viewDirectionW) {
+        float radius = projectedGroundInfos.x;
+        float height = projectedGroundInfos.y;
+
+        // reproject the cube ground to a sky sphere
+        // to help with shadows
+        // vec3 p = normalize(vPositionW);
+        vec3 camPos = vEyePosition.xyz;
```
[the remark this answers is not in the document]
Comment (sebavan) [reply packages/dev/core/src/Shaders/background.fragment.fx]: good catch, thanks

**Review by deltakosh — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14470/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**